docs/application-troubleshooting.md

```diff
@@ -0,0 +1,104 @@
+# Application Troubleshooting.
+
+This guide is to help users debug applications that are deployed into Kubernetes and not behaving correctly.
+This is *not* the guide for people who want to debug their cluster.  For that you should check out
+[this guide](cluster-troubleshooting.md)
+
+## FAQ
+Users are highly encouraged to checkout our [FAQ](https://github.com/GoogleCloudPlatform/kubernetes/wiki/User-FAQ)
+
+## Diagnosing the problem
+The first step in troubleshooting is triage.  What is the problem?  Is it your Pods, your Replication Controller or
+your Service?
+   * [Debugging Pods](#debugging-pods)
+   * [Debugging Replication Controllers](#debugging-replication-controllers)
+   * [Debugging Services](#debugging-services)
+
+### Debugging Pods
+The first step in debugging a Pod is taking a look at it.  For the purposes of example, imagine we have a pod
+```my-pod``` which holds two containers ```container-1``` and ```container-2```
+
+First, describe the pod.  This will show the current state of the Pod and recent events.
+
+```sh
+export POD_NAME=my-pod
+kubectl describe pods ${POD_NAME}
+```
+
+Look at the state of the containers in the pod.  Are they all ```Running```?  Have there been recent restarts?
+
+Let's suppose that ```container-2``` has been crash looping and you don't know why, you can take a look at the logs of
+the current container:
+
+```sh
+kubectl logs ${POD_NAME} ${CONTAINER_NAME}
+```
+
+If your container has previously crashed, you can access the previous containers crash log with:
+```sh
+kubectl logs --previous ${POD_NAME} ${CONTAINER_NAME}
+```
+
+Alternately, you can run commands inside that container with ```exec```:
+
+```sh
+kubectl exec ${POD_NAME} -c ${CONTAINER_NAME} -- ${CMD} ${ARG1} ${ARG2} ... ${ARGN}
+```
+
+If none of these approaches work, you can find the host machine that the pod is running on and SSH into that host,
+but this is generally not recommended, and indeed if you find yourself needing to ssh into a machine, please file a
+feature request on GitHub describing your use case and why these tools are insufficient.
```

Comment:
Nit: Suggest splitting this into two sentences.

Also, why is this not recommended? How about instead we say something like "this shouldn't be needed because of all the other awesome debugging tools we give you"?